autoload/ale/cursor.vim

```diff
@@ -95,26 +102,35 @@ function! ale#cursor#EchoCursorWarningWithDelay() abort
         return
     endif
 
-    call s:StopCursorTimer()
+    let l:buffer = bufnr('')
+
+    if ale#Var(l:buffer, 'echo_cursor')
+        call s:StopCursorTimer()
+
+        let l:pos = getcurpos()[0:2]
 
-    let l:pos = getcurpos()[0:2]
+        " Check the current buffer, line, and column number against the last
+        " recorded position. If the position has actually changed, *then*
+        " we should echo something. Otherwise we can end up doing processing
+        " the echo message far too frequently.
+        if l:pos != s:last_pos
+            let l:delay = ale#Var(bufnr(''), 'echo_delay')
 
-    " Check the current buffer, line, and column number against the last
-    " recorded position. If the position has actually changed, *then*
-    " we should echo something. Otherwise we can end up doing processing
-    " the echo message far too frequently.
-    if l:pos != s:last_pos
-        let l:delay = ale#Var(bufnr(''), 'echo_delay')
+            let s:last_pos = l:pos
+            let s:cursor_timer = timer_start(
+                        \   l:delay,
+                        \   function('ale#cursor#EchoCursorWarning')
+                        \)
+        endif
+    endif
 
-        let s:last_pos = l:pos
-        let s:cursor_timer = timer_start(
-        \   l:delay,
-        \   function('ale#cursor#EchoCursorWarning')
-        \)
+    if ale#Var(l:buffer, 'cursor_detail')
+        call ale#preview#CloseIfTypeMatches('ale-preview')
+        call ale#cursor#ShowCursorDetail({'stay_here': 1})
     endif
```

Comment:
You should remove the changes to this function, apart from the first line. What this function does is check if the cursor position has actually changed, and echo messages only after a short delay. If you don't wait for a short delay, the preview window will be closed and opened on every single keypress that moves the cursor, and that will slow down editing significantly.